python/pyspark/mllib/linalg.py

```diff
@@ -687,6 +702,100 @@ def __eq__(self, other):
                 all(self.values == other.values))
 
 
+class SparseMatrix(object):
+    """Sparse Matrix stored in CSC format."""
+    def __init__(self, numRows, numCols, colPtrs, rowIndices, values,
+                 isTransposed=False):
+        self.numRows = numRows
+        self.numCols = numCols
+        self.isTransposed = isTransposed
+        if isinstance(colPtrs, basestring):
+            self.colPtrs = np.frombuffer(colPtrs, dtype=np.uint64)
+        else:
+            self.colPtrs = np.asarray(colPtrs, dtype=np.uint64)
+
+        if self.isTransposed and self.colPtrs.size != numRows + 1:
```

Comment:
It is cleaner to write

~~~python
if self.isTranspose:
  ...
else:
  ...
~~~